membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -38,56 +39,53 @@ object AttributesFromZuora extends LoggingWithLogstashFields {
         })
       )
     } yield {
-      AttributesMaker.attributes(identityId, subscriptions, forDate)
+      AttributesMaker.zuoraAttributes(identityId, subscriptions, forDate)
     }
 
-
-    val attributesFromDynamo: Future[(String, Option[Attributes])] = dynamoAttributeService.get(identityId) map { attributes =>
+    val attributesFromDynamo: Future[Option[DynamoAttributes]] = dynamoAttributeService.get(identityId) map { attributes =>
       dynamoAttributeService.ttlAgeCheck(attributes, identityId, forDate)
-      ("Dynamo", attributes)
+      attributes
     }
 
-    val attributesFromZuora: Future[Disjunction[String, Option[Attributes]]] = attributesDisjunction.run.map {
+    val defaultIfZuoraFails: Future[Option[ZuoraAttributes]] = attributesFromDynamo map { maybeDynamoAttributes => maybeDynamoAttributes map {DynamoAttributes.asZuoraAttributes(_)}}
```

Comment:
It feels a bit weird that we're converting DynamoAttributes to ZuoraAttributes...

Reply:
Good point. Now I'm just doing this in the dynamoAndZuoraAgree comparison, but I think it makes sense in that context. Well, and the test.